### ISSUE TITLE
problems with exploding gradients in decision network

### ISSUE THREAD

**Issue by insunday** [skokec/segdec-net-jim2019#14]
Amazing algorithm. I am using this algorithm on my own dataset, there is around 600 data images and a third of them have defects. But I keep experiencing expolding gradients when training the decision network? Do you have this probelm, if so how did you solve it. Also I don't really understand why you split the data in to three folders. It that for validation or to get better results?

**Comment by skokec** [comment]
Hi, we have not experience issues with the exploding gradients. Since each feature layer is normalized to 0-mean and unit scale there should not be any problems with that. You might try to reduce the learning rate or applying ```tf.clip_by_norm ``` to see if this helps.

Data is split only due to small size of KolektorSDD dataset, which we could not split into train/test set without having issues with too few samples, so instead we used 3-fold cross-validation.

Best,
Domen
